x/exchange/orders_test.go

```diff
@@ -0,0 +1,2786 @@
+			var actual bool
+			testFunc := func() {
+				actual = tc.order.PartialFillAllowed()
+			}
+			require.NotPanics(t, testFunc, "PartialFillAllowed()")
+			assert.Equal(t, tc.exp, actual, "PartialFillAllowed() result")
+		})
+	}
+}
+
+func TestBidOrder_GetExternalID(t *testing.T) {
+	tests := []struct {
+		name  string
+		order BidOrder
+		exp   string
+	}{
+		{name: "empty", order: BidOrder{ExternalId: ""}, exp: ""},
+		{name: "something", order: BidOrder{ExternalId: "something"}, exp: "something"},
+		{name: "a uuid", order: BidOrder{ExternalId: "36585FC1-C11D-42A4-B1F7-92B0D8229BC7"}, exp: "36585FC1-C11D-42A4-B1F7-92B0D8229BC7"},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var actual string
+			testFunc := func() {
+				actual = tc.order.GetExternalID()
+			}
+			require.NotPanics(t, testFunc, "GetExternalID()")
+			assert.Equal(t, tc.exp, actual, "GetExternalID() result")
+		})
+	}
+}
+
+func TestBidOrder_GetOrderType(t *testing.T) {
+	expected := OrderTypeBid
+	order := BidOrder{}
+	var actual string
+	testFunc := func() {
+		actual = order.GetOrderType()
+	}
+	require.NotPanics(t, testFunc, "GetOrderType()")
+	assert.Equal(t, expected, actual, "GetOrderType() result")
+
+}
+
+func TestBidOrder_GetOrderTypeByte(t *testing.T) {
+	expected := OrderTypeByteBid
+	order := BidOrder{}
+	var actual byte
+	testFunc := func() {
+		actual = order.GetOrderTypeByte()
+	}
+	require.NotPanics(t, testFunc, "GetOrderTypeByte()")
+	assert.Equal(t, expected, actual, "GetOrderTypeByte() result")
+}
+
+func TestBidOrder_GetHoldAmount(t *testing.T) {
+	tests := []struct {
+		name  string
+		order BidOrder
+		exp   sdk.Coins
+	}{
+		{
+			name: "just price",
+			order: BidOrder{
+				Price: sdk.NewInt64Coin("cucumber", 8),
+			},
+			exp: sdk.NewCoins(sdk.NewInt64Coin("cucumber", 8)),
+		},
+		{
+			name: "price and settlement fee with shared denom",
+			order: BidOrder{
+				Price:               sdk.NewInt64Coin("cucumber", 8),
+				BuyerSettlementFees: sdk.NewCoins(sdk.NewInt64Coin("acorn", 5), sdk.NewInt64Coin("cucumber", 1)),
+			},
+			exp: sdk.NewCoins(
+				sdk.NewInt64Coin("acorn", 5),
+				sdk.NewInt64Coin("cucumber", 9),
+			),
+		},
+		{
+			name: "price and settlement fee with different denom",
+			order: BidOrder{
+				Price:               sdk.NewInt64Coin("cucumber", 8),
+				BuyerSettlementFees: sdk.NewCoins(sdk.NewInt64Coin("acorn", 5), sdk.NewInt64Coin("banana", 1)),
+			},
+			exp: sdk.NewCoins(
+				sdk.NewInt64Coin("acorn", 5),
+				sdk.NewInt64Coin("banana", 1),
+				sdk.NewInt64Coin("cucumber", 8),
+			),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var actual sdk.Coins
+			testFunc := func() {
+				actual = tc.order.GetHoldAmount()
+			}
+			require.NotPanics(t, testFunc, "GetHoldAmount()")
+			assert.Equal(t, tc.exp, actual, "GetHoldAmount() result")
+		})
+	}
+}
+
+func TestBidOrder_Validate(t *testing.T) {
+	coin := func(amount int64, denom string) sdk.Coin {
```

Comment:
Same as above. It would be helpful to reuse existing functions so we don't have to redefine this.

Reply:
They're redefined in each test because 
* I want the function's name to be short so that there's less noise to sift through while reading the test cases
* I don't want a `coin` function in the package since it's often handy to have a variable named `coin`. So if it's a package-level function, it needs to be longer.
* Different tests have different needs, and it's not always handy to have `(amount, denom)` for the args. E.g. maybe it's easier in some test to have a `(string)`.
* By defining it in the test, I don't have to have `t *testing.T` as an arg for any of them that need to run a check (e.g. make sure an error is nil).